docs/docs/encryption.md

```diff
@@ -0,0 +1,53 @@
+### Using encryption
+***
+You can encrypt your data using your own encryption and secret keys from the command-line using a AES/CBC/PKCS5PADDING cipher.
+Values are first encrypted and then base64 encoded
+
+option                                        | description
+--------------------------------------------- | -----------
+--encrypt_data                                | encrypt the data field
+--encrypt_all                                 | encrypt the entire payload, takes precedence over encrypt_data if both specified
+--encryption_key                              | encryption key to be used
+--secret_key                                  | secret key to be used
+
+### Decryption
+***
+To decrypt your data you must first decode the string from base64 and then apply the cipher to decrypt. The method is provided in RowEncrypt.decrypt(), it looks like this:
+
+```
+try {
+    IvParameterSpec ivSpec = new IvParameterSpec(initVector.getBytes("UTF-8"));
+    SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
+
+    Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+    cipher.init(Cipher.DECRYPT_MODE, skeySpec, ivSpec);
+
+    return new String(cipher.doFinal(Base64.decodeBase64(value.getBytes("UTF-8"))), Charset.forName("UTF-8"));
+} catch (Exception ex) {
+    ex.printStackTrace();
+}
+```
+
+### Examples
+***
+
+insert into minimal set account_id =1, text_field='hello'
+
+Not encrypted
+```
+{"database":"shard_1","table":"minimal","type":"insert","ts":1490115785,"xid":153,"commit":true,"data":{"id":1,"account_id":1,"text_field":"hello"}}
+```
+
+--encrypt_data=true
+--encryption_key=aaaaaaaaaaaaaaaa
+--secret_key=RandomInitVector
```

Comment:
I'd like to push back on this, I think having the hash function is dangerous and since most decryption won't be on Maxwell they would then need the same hash elsewhere. A 16 byte key is pretty standard for encryption IMO, like tim said above I think a future 'encryption_scheme' option would be the best idea if someone doesn't like this algorithm.